beginner_source/blitz/tensor_tutorial.py

```diff
@@ -1,195 +1,189 @@
-# -*- coding: utf-8 -*-
 """
-What is PyTorch?
-================
-
-It’s a Python-based scientific computing package targeted at two sets of
-audiences:
-
--  A replacement for NumPy to use the power of GPUs
--  a deep learning research platform that provides maximum flexibility
-   and speed
+Tensors
+--------------------------------------------
 
-Getting Started
----------------
+Tensors are a specialized data structure that are very similar to arrays
+and matrices. In PyTorch, we use tensors to encode the inputs and
+outputs of a model, as well as the model’s parameters.
 
-Tensors
-^^^^^^^
+Tensors are similar to NumPy’s ndarrays, except that tensors can run on
+a GPU to accelerate computing. If you’re familiar with ndarrays, you’ll
+be right at home with the Tensor API. If not, follow along in this quick
+API walkthrough.
 
-Tensors are similar to NumPy’s ndarrays, with the addition being that
-Tensors can also be used on a GPU to accelerate computing.
 """
 
-from __future__ import print_function
 import torch
+import numpy as np
 
-###############################################################
-# .. note::
-#     An uninitialized matrix is declared,
-#     but does not contain definite known
-#     values before it is used. When an
-#     uninitialized matrix is created,
-#     whatever values were in the allocated
-#     memory at the time will appear as the initial values.
-
-###############################################################
-# Construct a 5x3 matrix, uninitialized:
 
-x = torch.empty(5, 3)
-print(x)
- 
-###############################################################
-# Construct a randomly initialized matrix:
+######################################################################
+# Tensor Initialization
+# ~~~~~~~~~~~~~~~~~~~~~
+#
+# Tensors can be initialized in various ways. Take a look at the following examples
```

Comment:
each introduction of a concept followed by a code snippet. one at a time.
Cleanly separating introducing concepts is way more important than minimizing code.

--------------

Tensors can be initialized in various ways. Take a look at the following examples

Directly from data

```
data = [[1, 2],[3, 4]]
x_data = torch.tensor(data)
print(x_data)
```

From a NumPy array:

```
x_np = np.array(data)
x_data = torch.from_numpy(x_np)
print(x_data)
```